401d56/code-challenges/gradle/src/main/java/hashtable/Hashtable.java

```diff
@@ -0,0 +1,56 @@
+package hashtable;
+
+public class Hashtable {
+
+    private Node[] map;
+
+    public Hashtable(int size){
+        map = new Node[size];
+    }
+
+    int hash(String key){
+        int hashValue = 0;
+        char[] letters = key.toCharArray();
+        for(int i = 0; i < letters.length; i++){
+            hashValue += letters[i];
+        }
+        hashValue = (hashValue * 843) % map.length;
+        return hashValue;
+    }
+
+    public void set(String key, String value) {
+        int hashkey = hash(key);
+
+        if (this.map[hashkey] == null) {
+            map[hashkey] = new Node(key, value);
+        } else {
+            Node temp = map[hashkey];
+            map[hashkey] = new Node(key, value);
+            map[hashkey].setNext(temp);
+        }
+    }
+
+    // get
+    public String get(String key){
+        int hashkey = hash(key);
+        String value = map[hashkey].getValue();
+
+        return value;
+    }
+
+    // getLength
+    public int getLength(){
+        int value = map.length;
+        return value;
+    }
+
+    // contains
+    public Boolean contains(String key){
```

Comment:
A bucket can have multiple things in it, it's why we use things like linked lists to hold the items in those buckets. This will return true if a key would go to a bucket that already has something in it, but isn't there. You need to check through the nodes in the bucket